deltachat-ios/Controller/WebViewViewController.swift

```diff
@@ -192,6 +202,47 @@ class WebViewViewController: UIViewController, WKNavigationDelegate {
         webView.evaluateJavaScript("WKWebView_SearchPrev()", completionHandler: nil)
         updateAccessoryBar()
     }
+
+    func openChatFor(url: URL) {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
+              let emailAddress = parseEmailAddress(from: url) else {
+            return
+        }
+
+        // FIXME: lookupContactIdByAddress is still broken
+        // let contactId = dcContext.lookupContactIdByAddress(emailAddress)
+
+        // workaround:
+        let contacts: [Int] = dcContext.getContacts(flags: DC_GCL_ADD_SELF, queryString: emailAddress)
+        let index = contacts.firstIndex(where: { dcContext.getContact(id: $0).email == emailAddress }) ?? -1
+        var contactId = 0
+        if index >= 0 {
+            contactId = contacts[index]
+        }
```

Comment:
```suggestion
        let contactId = index >= 0 ? contacts[index] : 0
```

unmutable is always better :)